src/test/javascript/spec/component/course/course-overview.component.spec.ts

```diff
@@ -149,7 +151,7 @@ describe('CourseOverviewComponent', () => {
                 { provide: ActivatedRoute, useValue: { params: of({ courseId: course1.id }), snapshot: { firstChild: { routeConfig: { path: 'courses/1/exercises' } } } } },
```

Comment:
These should be all the providers you need to mock:
```
        private courseService: CourseManagementService,
        private courseExerciseService: CourseExerciseService,
        private courseCalculationService: CourseScoreCalculationService,
        private learningGoalService: LearningGoalService,
        private route: ActivatedRoute,
        private teamService: TeamService,
        private jhiWebsocketService: JhiWebsocketService,
        private serverDateService: ArtemisServerDateService,
        private alertService: AlertService,
        private changeDetectorRef: ChangeDetectorRef,
        private profileService: ProfileService,
        private tutorialGroupService: TutorialGroupsService,
        private tutorialGroupsConfigurationService: TutorialGroupsConfigurationService,
        private metisConversationService: MetisConversationService,
        private router: Router,
```

Reply:
It was a bit more difficult here since there was a lot of code that I didn't write myself, but it's cleaner now!